// bearychat/dist/channel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _channel = require('./channel.archive');

var _channel2 = _interopRequireDefault(_channel);

var _channel3 = require('./channel.unarchive');

var _channel4 = _interopRequireDefault(_channel3);

var _channel5 = require('./channel.create');

var _channel6 = _interopRequireDefault(_channel5);

var _channel7 = require('./channel.info');

var _channel8 = _interopRequireDefault(_channel7);

var _channel9 = require('./channel.invite');

var _channel10 = _interopRequireDefault(_channel9);

var _channel11 = require('./channel.join');

var _channel12 = _interopRequireDefault(_channel11);

var _channel13 = require('./channel.kick');

var _channel14 = _interopRequireDefault(_channel13);

var _channel15 = require('./channel.leave');

var _channel16 = _interopRequireDefault(_channel15);

var _channel17 = require('./channel.list');

var _channel18 = _interopRequireDefault(_channel17);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  archive: _channel2.default,
  unarchive: _channel4.default,
  create: _channel6.default,
  info: _channel8.default,
  invite: _channel10.default,
  join: _channel12.default,
  kick: _channel14.default,
  leave: _channel16.default,
  list: _channel18.default
};
module.exports = exports['default'];
