// bearychat/dist/HTTPClient.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _setInPath = require('./_setInPath');

var _setInPath2 = _interopRequireDefault(_setInPath);

var _validation = require('./_validation.json');

var _validation2 = _interopRequireDefault(_validation);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ResponseError = function (_Error) {
  _inherits(ResponseError, _Error);

  function ResponseError(message, response) {
    _classCallCheck(this, ResponseError);

    var _this = _possibleConstructorReturn(this, (ResponseError.__proto__ || Object.getPrototypeOf(ResponseError)).call(this, message));

    _this.response = response;
    return _this;
  }

  return ResponseError;
}(Error);

var UnexpectedCodeError = function (_Error2) {
  _inherits(UnexpectedCodeError, _Error2);

  function UnexpectedCodeError(message, code) {
    _classCallCheck(this, UnexpectedCodeError);

    var _this2 = _possibleConstructorReturn(this, (UnexpectedCodeError.__proto__ || Object.getPrototypeOf(UnexpectedCodeError)).call(this, message));

    _this2.code = code;
    return _this2;
  }

  return UnexpectedCodeError;
}(Error);

function tryParseResponse(response) {
  var shouldParseAsJson = response.headers.has('Content-Type') && response.headers.get('Content-Type').indexOf('application/json') === 0;
  var bodyPromise = shouldParseAsJson ? response.json() : response.text();
  return bodyPromise.then(function (data) {
    return {
      data: data,
      response: response
    };
  }).catch(function () {
    return {
      // throw ResponseError in case response body parse error.
      error: new ResponseError(response.statusText, response)
    };
  });
}

function callMethod(method, payload) {
  var context = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : this;

  return new Promise(function (resolve, reject) {
    method.call(context, payload).then(tryParseResponse).then(function (_ref) {
      var data = _ref.data,
          response = _ref.response,
          error = _ref.error;

      if (error) {
        reject(error);
      } else if (typeof data === 'string') {
        resolve(data);
      } else if (typeof data.code === 'number' && data.code !== 0) {
        reject(new UnexpectedCodeError(data.error, data.code));
      } else if (response.status < 200 || response.status >= 300) {
        reject(new ResponseError(response.statusText, response));
      } else {
        resolve(data);
      }
    });
  });
}

function getWrappedMethod(path, token) {
  // eslint-disable-next-line import/no-dynamic-require,global-require
  var method = require('.' + path);
  var settings = _validation2.default[path];
  settings = settings[Object.keys(settings)[0]];
  var needAuth = settings.authentication !== false;
  // eslint-disable-next-line func-names
  return function (payload) {
    if (needAuth) {
      payload = Object.assign({}, payload, {
        token: token
      });
    }
    return callMethod(method, payload, this);
  };
}

var HTTPClient = function HTTPClient(token) {
  var _this3 = this;

  _classCallCheck(this, HTTPClient);

  Object.keys(_validation2.default).forEach(function (path) {
    var setPath = path.replace(/^\//, '');
    (0, _setInPath2.default)(_this3, setPath, getWrappedMethod(path, token));
  });
};

exports.default = HTTPClient;
module.exports = exports['default'];
