// bearychat/dist/incoming.builder.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Builder = function () {
  function Builder() {
    _classCallCheck(this, Builder);

    this.reset();
  }

  _createClass(Builder, [{
    key: 'reset',
    value: function reset() {
      this.text = null;
      this.markdown = false;
      this.channel = null;
      this.attachments = [];
    }
  }, {
    key: 'to',
    value: function to(channel) {
      this.channel = channel;

      return this;
    }
  }, {
    key: 'withText',
    value: function withText(text) {
      var useMarkdown = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

      this.text = text;
      this.markdown = useMarkdown;

      return this;
    }
  }, {
    key: 'withMarkdown',
    value: function withMarkdown(text) {
      return this.withText(text, true);
    }
  }, {
    key: 'withAttachment',
    value: function withAttachment(attachment) {
      this.attachments.push(attachment);

      return this;
    }
  }, {
    key: 'build',
    value: function build() {
      var message = {
        text: this.text,
        markdown: this.markdown
      };

      if (this.channel) {
        message.channel = this.channel;
      }

      if (this.attachments.length > 0) {
        message.attachments = [];
        this.attachments.forEach(function (attachment) {
          if (!attachment.title && !attachment.text) {
            throw new Error('attachment title or text is required');
          }
          message.attachments.push(attachment);
        });
      }

      return message;
    }
  }]);

  return Builder;
}();

exports.default = {
  withText: function withText(text, useMarkdown) {
    return new Builder().withText(text, useMarkdown);
  },
  withMarkdown: function withMarkdown(text) {
    return new Builder().withMarkdown(text);
  }
};
module.exports = exports['default'];
